include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -135,6 +140,280 @@ class alignment_file_format_sam
+              ref_seq_type                                            & SEQAN3_DOXYGEN_ONLY(ref_seq),
+              ref_id_type                                             & ref_id,
+              ref_offset_type                                         & ref_offset,
+              align_type                                              & align,
+              flag_type                                               & flag,
+              mapq_type                                               & mapq,
+              mate_type                                               & mate,
+              tag_dict_type                                           & tag_dict,
+              e_value_type                                            & SEQAN3_DOXYGEN_ONLY(e_value),
+              bit_score_type                                          & SEQAN3_DOXYGEN_ONLY(bit_score))
+    {
+        static_assert(detail::decays_to_ignore_v<ref_offset_type> ||
+                      detail::is_type_specialisation_of_v<ref_offset_type, std::optional>);
+
+        using stream_buf_t = std::istreambuf_iterator<typename stream_type::char_type>;
+        auto stream_view = view::subrange<decltype(stream_buf_t{stream}), decltype(stream_buf_t{})>
+                               {stream_buf_t{stream}, stream_buf_t{}};
+
+        auto next_field = view::take_until_or_throw(is_char<'\t'>);
+
+        // these variables need to be stored to compute the ALIGNMENT
+        std::conditional_t<!detail::decays_to_ignore_v<ref_offset_type>,
+                           ref_offset_type,
+                           std::optional<size_t>> ref_offset_tmp{};
+        [[maybe_unused]] std::string ref_id_tmp{}; // in case reference information is given
+        [[maybe_unused]] std::conditional_t<!detail::decays_to_ignore_v<offset_type>, offset_type, size_t> offset_tmp{};
+        [[maybe_unused]] std::conditional_t<!detail::decays_to_ignore_v<offset_type>, offset_type, size_t> soft_clipping_end{};
+        [[maybe_unused]] std::vector<std::pair<char, size_t>> cigar{};
+        [[maybe_unused]] size_t ref_length{0}, seq_length{0}; // length of aligned part for ref and query
+
+        // Header
+        // -------------------------------------------------------------------------------------------------------------
+        if (is_char<'@'>(*std::ranges::begin(stream_view))) // we always read the header if present
+        {
+            assert(header_ptr != nullptr);
+            read_header(stream_view, header_ptr);
+        }
+
+        // Fields 1-5: ID FLAG REF_ID REF_OFFSET MAPQ
+        // -------------------------------------------------------------------------------------------------------------
+        read_field(stream_view | next_field, id);
+        std::ranges::next(std::ranges::begin(stream_view)); // skip tab
+
+        read_field(stream_view | next_field, flag);
+        std::ranges::next(std::ranges::begin(stream_view));
```

Comment:
here std::ranges::next works, but only because it's an single-pass range. This is not required to work, though, and will change when some views become move-only. Please use std::ranges:.advance in these places.

Reply:
Please make a card for this in the range module or i/o module that we should use istream_range and advance in the future,